sdk/samples/iot/provisioning/README.md

```diff
@@ -11,50 +11,129 @@ urlFragment: iot-provisioning-samples
+
+1. Enter the directory `/azure-sdk-for-c/sdk/samples/iot/provisioning/src/` and run the script using the following form:
+
+    Linux:
+    ```bash
+    ./generate_certificate.sh
+    ```
+    Windows:
+    ```cmd
+    generate_certificates.cmd
+    ```
+2. Set the environment variable `AZ_IOT_DEVICE_X509_CERT_PEM_FILE` noted at the bottom of the output.
+3. In your Azure DPS, add a new individual device enrollment using the recently generated `device_ec_cert.pem` file.  See [here](https://docs.microsoft.com/en-us/azure/iot-dps/quick-create-simulated-device-x509#create-a-device-enrollment-entry-in-the-portal) for further instruction.  After creation, the Registration ID of your device should appear as `paho-sample-device1` in the Individual Enrollments tab.
+
+### Paho IoT Provisioning (SAS)
+This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/provisioning/src/paho_iot_provisioning_sas_example.c) uses SAS symmetric key authentication to connect to Azure IoT Hub Device Provisioning Service (DPS). When enrolling a device with the DPS, an option to auto-generate the key will be provided.  
+
+* In your Azure DPS, add a new individual device enrollment using SAS. See [here](https://docs.microsoft.com/en-us/azure/iot-dps/quick-create-simulated-device-symm-key#create-a-device-enrollment-entry-in-the-portal) for further instruction. After creation, the Registration ID of your device will appear in the Individual Enrollments tab.
+
+
+
+## Build and Run the Sample
+
+1. Set the remaining environment variables.  Setting a variable will take the following form:
+
+	  Linux:
+	  ```bash
+	  export ENV_VARIABLE_NAME=VALUE
+	  ```
+	  Windows:
+	  ```cmd
+	  set "ENV_VARIABLE_NAME=VALUE"
+	  ```
+	#### Paho IoT Provisioning (Certificates)
+  * `VCPKG_DEFAULT_TRIPLET` and `VCPKG_ROOT`: These should already be set per these [directions](https://github.com/Azure/azure-sdk-for-c#development-environment).
+  * `AZ_IOT_DEVICE_X509_CERT_PEM_FILE`: This should already be set per directions above.
+  * `AZ_IOT_ID_SCOPE`: Copy this value from the Overview tab in your Azure DPS.
+  * `AZ_IOT_REGISTRATION_ID`: This should be `paho-sample-device1`.
+	#### Paho IoT Provisioning (SAS)
+  * `VCPKG_DEFAULT_TRIPLET` and `VCPKG_ROOT`: These should already be set per these [directions](https://github.com/Azure/azure-sdk-for-c#development-environment).
+  * `AZ_IOT_PROVISIONING_SAS_KEY`: Select your enrolled device from the Individual Enrollments tab and copy its Primary Key.
+  * `AZ_IOT_ID_SCOPE`: Copy this value from the Overview tab in your Azure DPS.
+  * `AZ_IOT_REGISTRATION_ID_SAS`: Copy the Registration Id of your SAS device from the Individual Enrollments tab.
+
+
+2. Compile the code:
+  * Enter the directory `/azure-sdk-for-c/cmake`.  If it does not exist, please create it.
+  * Build the directory structure:
+
+    ```bash
+    cmake -DTRANSPORT_PAHO=ON ..
+    ```
+  * Build the sample:
+
+    Linux:
+    ```bash
+    make
```

Comment:
nit: I think this should stay `cmake --build .` It does the same thing as `make` and prevents confusion if someone decides to use Ninja etc.

Reply:
> Great work here. I'm usually pretty "nit"y with docs but this is awesome work. Thanks for all the docs!

Thanks Dane, I appreciate that!